# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLBlitHelper.h
# Commit: 6149f5b15c5e
# Full Hash: 6149f5b15c5e278ef15ea36fa896fa8c146ebdca
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2021-01-14 08:32:45
# Regressor Bug: 1640607
# File Overlap Count: 1
# Description:
#   Bug 1640607 - Send SurfaceDescriptors for GPU blitting for video-to-webgl. r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D101061
# ==============================================================================

diff -r 29fd3a6a7e27 -r 6149f5b15c5e gfx/gl/GLBlitHelper.h
--- a/gfx/gl/GLBlitHelper.h	Wed Jan 13 20:14:54 2021 +0000
+++ b/gfx/gl/GLBlitHelper.h	Wed Jan 13 21:49:40 2021 +0000
@@ -39,6 +39,7 @@
 class PlanarYCbCrImage;
 class SurfaceTextureImage;
 class MacIOSurfaceImage;
+class SurfaceDescriptor;
 class SurfaceDescriptorD3D10;
 class SurfaceDescriptorDXGIYCbCr;
 }  // namespace layers
@@ -194,6 +195,8 @@
   bool BlitImageToFramebuffer(layers::Image* srcImage,
                               const gfx::IntSize& destSize,
                               OriginPos destOrigin);
+  bool BlitSdToFramebuffer(const layers::SurfaceDescriptor&,
+                           const gfx::IntSize& destSize, OriginPos destOrigin);
 
  private:
   bool BlitImage(layers::GPUVideoImage* srcImage, const gfx::IntSize& destSize,